### ISSUE TITLE
Linting Errors Still Throw Exit Code 0

### ISSUE THREAD

**Issue by magnanimus1994** [eslint/eslint#12366]
**Context**
We're using a custom eslint plugin to run specific rules against browser automation tests with TestCafe. 

**Commands Run**
```bash
npm i eslint babel-eslint eslint-plugin-prodperfect-testcafe
find tests/ -name "*.js" ! -name tracking-verification.js -exec node_modules/.bin/eslint '{}' -c .eslintrc.json ';'
```

**ESLint Version:** 6.5.1
**Node Version:** 8.9.4
**npm Version:** 5.6.0

**Parser**
babel-eslint@10.0.3

**Configuration**
```js
{
   "parser": "babel-eslint",
   "parserOptions": {
   	"ecmaVersion": 6,
   	"sourceType": "module"
   },
   "plugins": [
   	"prodperfect-testcafe"
   ],
   "rules": {
     "prodperfect-testcafe/no-debug-statements":"error",
     "prodperfect-testcafe/no-only-statements":"error",
     "prodperfect-testcafe/p2-test-names":"error"
   }
}
```

**Expected outcome**
Should throw exit status 1

**Actual output**
```
Exit code: 0
#!/bin/bash -eo pipefail

/home/circleci/project/tests/regression-tests.js
  126:1  error  Unexpected .only() - Please remove before pushing to production   prodperfect-testcafe/no-only-statements
  448:1  error  Test names should end in a properly formatted v4 UUID             prodperfect-testcafe/p2-test-names
  455:9  error  Unexpected .debug() - Please remove before pushing to production  prodperfect-testcafe/no-debug-statements
  480:1  error  Test names should end in a properly formatted v4 UUID             prodperfect-testcafe/p2-test-names

✖ 4 problems (4 errors, 0 warnings)
```

**Comment by g-plane** [comment]
Thanks for this issue.

It seems that you were running ESLint by a custom JavaScript file you wrote, right?

**Comment by magnanimus1994** [comment]
Yes. Here are the contents of the `index.js` of eslint plugin: 

```js
module.exports = {
  rules: {
    "no-debug-statements": {
      meta: {
        docs: {
          description: "disallow .debug() statements",
          category: "Possible Errors",
          recommended: true
        },
        fixable: null,
        schema: [],
        messages: {
            unexpected: "Unexpected .debug() - Please remove before pushing to production."
        }
      },
      create: function(context) {
        return {
          CallExpression(node) {
	    const property = node.callee.property || {type: null, name: null};
            if (property.type === 'Identifier' && property.name === 'debug') {
              context.report({node, messageId: "unexpected"});
            }
          }
        };
      }
    },
    "no-only-statements": {
      meta: {
        docs: {
          description: "disallow test.only() statements",
          category: "Possible Errors",
          recommended: true
        },
        fixable: null,
        schema: [],
        messages: {
            unexpected: "Unexpected .only() - Please remove before pushing to production."
        }
      },
      create: function(context) {
        return {
          CallExpression(node) {
            const object = node.callee.object || {type: null, name: null};
            const property = node.callee.property || {type: null, name: null};
            if (
                object.type === 'Identifier' && 
                object.name === 'test' &&
                property.type === 'Identifier' && 
                property.name === 'only'
            ) {
              context.report({node, messageId: "unexpected"});
            }
          }
        };
      }
    },
    "p2-test-names": {
      meta: {
        docs: {
          description: "ensure proper formatting of test names",
          category: "Best Practices",
          recommended: true
        },
        fixable: null,
        schema: [],
        messages: {
          capitals: "Test name should not contain capitals.",
          doubleUnderscores: "Test name should not contain double underscores.",
          longDescription: "Test description should be less than 8 words.",
          nameStart: "Test name should start with the word \'test\'.",
          noDescription: "Test name needs a description.",
          testNum: "Test needs a number after the word \'test\'.",
          underscores: "Test name components should be separated by underscores.",
          uuid: "Test names should end in a properly formatted v4 UUID."
        }
      },
      create: function(context) {
        return {
          CallExpression(node) {
            if (node.callee.type === 'Identifier' && node.callee.name === 'test') {
              if (node.arguments[0].value.match(/\s/)) {
                context.report({node, messageId: "underscores"});
              } else {
                let name = node.arguments[0].value.split('_');

                // Check all lowercase and proper underscores
                name.forEach(str => {
                  if (!str) {
                    context.report({node, messageId: "doubleUnderscores"});
                  } else if (/[A-Z]/.test(str)) {
                    context.report({node, messageId: "capitals"});
                  }
                });

                // Check v4 UUID appended at the end
                const uuid = name.splice(-1)[0] || '';
                if (!uuid.match(/^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/i)) {
                  context.report({node, messageId: "uuid"});
                }

                // check description is between 0 and 8 words
                const description = name.splice(2) || [];
                if (description.length === 0) {
                  context.report({node, messageId: "noDescription"});
                } else if (description.length > 7) {
                  context.report({node, messageId: "longDescription"});
                }

                // Check that test begins with 'test' and then the test number
                if (!name[0].match(/test/i)) {
                  context.report({node, messageId: "nameStart"});
                }
                
                const testNum = parseInt(name[1], 10);
                if (isNaN(testNum) || testNum < 1) {
                  context.report({node, messageId: "testNum"});
                }
              }
            }
          }
        }
      }
    }
  }
};
```

**Comment by platinumazure** [comment]
Hi @magnanimus1994, thanks for the issue.

I think this is as expected, based on the [GNU `find(1)` man page](http://man7.org/linux/man-pages/man1/find.1.html). Specifically, the Exit Status section notes that the exit status of the `find` command is zero if all files are processed successfully by `find`, nonzero otherwise. The `-exec` command also documents that it runs `command ; true` which returns zero exit status no matter what.

You could try using the `-exec` with `+` terminator, which appends the found files to a single command, although I don't know if that will let the ESLint exit status go through. Otherwise, I recommend piping the output of `find` so that it goes to ESLint properly (maybe through `xargs`).

In short, this is not an ESLint issue and `find` is doing what it's supposed to, as far as I can tell.

**Comment by magnanimus1994** [comment]
Thank you, @platinumazure!  `find tests/ -name "*.js" ! -name tracking-verification.js -print0 | xargs -0 node_modules/.bin/eslint -c .eslintrc.json` works!
